aws/data_source_aws_sfn_state_machine_test.go

```diff
@@ -0,0 +1,105 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccDataSourceAwsSfnStateMachine(t *testing.T) {
+	rName := acctest.RandString(5)
+	resourceName := "data.aws_sfn_state_machine.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsSfnStateMachineConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceAwsSfnStateMachineCheck(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "role_arn"),
+					resource.TestCheckResourceAttrSet(resourceName, "definition"),
+					resource.TestCheckResourceAttrSet(resourceName, "creation_date"),
+					resource.TestMatchResourceAttr(resourceName, "creation_date", regexp.MustCompile("^20[0-9]{2}-")),
+					resource.TestCheckResourceAttr(resourceName, "status", "ACTIVE"),
```

Comment:
This data source testing can be more accurate to ensure the data source and resource values match by using `resource.TestCheckResourceAttrPair()`:

```go
					resource.TestCheckResourceAttrPair(resourceName, "arn", dataSourceName, "arn"),
					resource.TestCheckResourceAttrPair(resourceName, "creation_date", dataSourceName, "creation_date"),
					resource.TestCheckResourceAttrPair(resourceName, "definition", dataSourceName, "definition"),
					resource.TestCheckResourceAttrPair(resourceName, "name", dataSourceName, "name"),
					resource.TestCheckResourceAttrPair(resourceName, "role_arn", dataSourceName, "role_arn"),
					resource.TestCheckResourceAttrPair(resourceName, "status", dataSourceName, "status"),
```

Reply:
added - cheers